### ISSUE TITLE
expo web browser: swipe right on iOS close browser

### ISSUE THREAD

**Issue by sa1dai** [expo/expo#6918]
### Environment

Expo 36, managed workflow, iOS

### Steps to Reproduce

1. Open snack at link in Reproducible demo
2. Select Expo 36 at the bottom
3. Click save
4. Open on expo client on iOS
5. Open snack in saved snacks
6. Click 'Open web browser'
7. Swipe right 

### Expected Behavior

Swipe right shouldn't close browser and it doesn't close if choose Expo 35 in step 2 above

### Actual Behavior

Swipe right closes browser

### Reproducible Demo

https://snack.expo.io/@blackterminalapp/basic-webbrowser-usage?platform=android&name=Basic%20WebBrowser%20usage&sdkVersion=36.0.0&dependencies=expo-web-browser&sourceUrl=https%3A%2F%2Fdocs.expo.io%2Fstatic%2Fexamples%2Fv36.0.0%2Fweb-browser.js

**Comment by cruzach** [comment]
What device/iOS version are you using? I tried the snack on both SDK 35 and 36, and swiping right didn't close the WebBrowser

**Comment by sa1dai** [comment]
@cruzach 
I have created blank expo project, paste code from snack.
Expo diagnostics:
Expo CLI 3.11.7 environment info:
    System:
      OS: Linux 5.3 Ubuntu 18.04.3 LTS (Bionic Beaver)
      Shell: 4.4.20 - /bin/bash
    Binaries:
      Node: 10.18.0 - /usr/bin/node
      npm: 6.13.4 - /usr/bin/npm
    npmPackages:
      expo: ~36.0.0 => 36.0.2 
      react: ~16.9.0 => 16.9.0 
      react-native: https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz => 0.61.4 
    npmGlobalPackages:
      expo-cli: 3.11.7

Then I tested at this two devices: one with iOS 12.4.2 and one with iOS 13.3.
And on both I can close browser with swipe right.

<a href="https://imgflip.com/gif/3nr09f"><img src="https://i.imgflip.com/3nr09f.gif" title="made at imgflip.com"/></a>

**Comment by cruzach** [comment]
You're right, it happens on a local project just not in snack. Thanks for specifying!

**Comment by cruzach** [comment]
Seems like this was the only major change for `expo-web-browser` fro SDK 36- https://github.com/expo/expo/pull/6345

**Comment by Ciberusps** [comment]
Same problem, after close webBrowser via swiper right its `can't be opened anymore`, because swipe right - dont "dismissBrowser".

**Comment by joshg2k** [comment]
Is this issue resolved as part of Expo SDK 37? I can't tell from https://github.com/orgs/expo/projects/5

**Comment by joshg2k** [comment]
In case anybody needs a workaround until this issue is resolved, here's an example of how you can programmatically dismiss the browser when trying to re-open the WebBrowser after a swipe dismiss:

```
_webBrowserRetry = false;

_handleWebLink = link => {
        WebBrowser.openBrowserAsync(link).then(result => {
            // Successfully closed WebBrowser, either by clicking done or programmatically dismissing.
            this._webBrowserRetry = false;
        }).catch(err => {
            if (err.message == 'Another WebBrowser is already being presented.') {
                WebBrowser.dismissBrowser();

                // Limit retries so we don't end up with an infinite recursion.
                if (!this._webBrowserRetry) {
                    this._webBrowserRetry = true;
                    this._handleWebLink(link);
                }
            } else {
                throw err;
            }
        });
}
```

**Comment by hassey13** [comment]
@joshg2k did you find that snippet to work? I find the `dismissBrowser` does not dismissed the swipe closed browser on iPhone 6 running iOS 12.4

**Comment by cjinghong** [comment]
I'm using Expo SDK version 36, and I'm facing the same exact problem. The workaround does not work completely though, after explicitly dismissing the WebBrowser for the first time, the function would have to be called twice to open the WebBrowser again. 

Currently, what I did was to omit the limiting of retries completely, then it works as expected.

But it's pretty scaring looking at a potential infinite loop if it were to throw the same error twice...

**Comment by shaheryar-ahmad** [comment]
I was having the same issue, came up with an approach that works for me. Give it try :)

`
 _handlePressButtonAsync = async (items) => {

    await WebBrowser.openBrowserAsync(items.link)
      .then(res => {
        console.log(res);
      })
      .catch(async err => {
        if (err.message === "Another WebBrowser is already being presented.") {

          WebBrowser.dismissBrowser();
          await WebBrowser.openBrowserAsync(items.link)
            .then(response => {
              console.log(response);
            })
            .catch(async error => {
              alert("something went wrong");
            });
        } else {
          alert("something went wrong");
        }
      });
  };
`

**Comment by joshg2k** [comment]
> @joshg2k did you find that snippet to work? I find the `dismissBrowser` does not dismissed the swipe closed browser on iPhone 6 running iOS 12.4

I don't recall testing it with this device and version, so I can't say for sure. That being said, Expo SDK 37 addresses this - although it appears to have disabled the swipe to dismiss functionality.

@axeldelafosse - Was it your intention to disable swipe to dismiss functionality in your fix?
